### DISCUSSION TITLE
Get file list of a project

### DISCUSSION THREAD

**Discussion by excalibur1983** [zapadi/redmine-net-api#305] [Q&A]
What's the correct approach to get the file lists corresponding to a specific project?
In other words, what's the example code for the following RESTful call using redmine-net-api?

    GET /projects/1/files.xml

**Comment by zapadi** [comment]
> What's the correct approach to get the file lists corresponding to a specific project? In other words, what's the example code for the following RESTful call using redmine-net-api?
> 
> ```
> GET /projects/1/files.xml
> ```

`
redmineManager.GetObjects<File>(new NameValueCollection(){{RedmineKeys.PROJECT_ID,"<projectId>"}});
`

**Reply by AlessandroNastasi92** [reply]
> > What's the correct approach to get the file lists corresponding to a specific project? In other words, what's the example code for the following RESTful call using redmine-net-api?
> > ```
> > GET /projects/1/files.xml
> > ```
> 
> `redmineManager.GetObjects<File>(new NameValueCollection(){{RedmineKeys.PROJECT_ID,"<projectId>"}});`

I'm trying with the exact same code and an exception raises:

_System.Collections.Generic.KeyNotFoundException: 'File'_

This is the stacktrace:

_in Redmine.Net.Api.Internals.UrlHelper.GetListUrl[T](RedmineManager redmineManager, NameValueCollection parameters)
   in Redmine.Net.Api.RedmineManager.GetPaginatedObjects[T](NameValueCollection parameters)
   in Redmine.Net.Api.RedmineManager.GetObjects[T](NameValueCollection parameters)
   in LoginWindow.MainWindow.LoginButtonClick(Object sender, RoutedEventArgs e) in C:\Users\xxxx\source\repos\xxxxt\xxxxt\LoginWindow.xaml.cs: riga 48
   in System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   in System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   in System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   in System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)
   in System.Windows.Controls.Primitives.ButtonBase.OnClick()
   in System.Windows.Controls.Button.OnClick()
   in System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   in System.Windows.UIElement.OnMouseLeftButtonUpThunk(Object sender, MouseButtonEventArgs e)
   in System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   in System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   in System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   in System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   in System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   in System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
   in System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   in System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   in System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   in System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   in System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   in System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   in System.Windows.UIElement.RaiseEvent(RoutedEventArgs args, Boolean trusted)
   in System.Windows.Input.InputManager.ProcessStagingArea()
   in System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   in System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   in System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   in System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   in System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   in MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   in MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   in System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   in System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   in System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   in MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   in MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   in System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   in System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   in System.Windows.Application.RunDispatcher(Object ignore)
   in System.Windows.Application.RunInternal(Window window)
   in System.Windows.Application.Run(Window window)
   in System.Windows.Application.Run()
   in LoginWindow.App.Main()_

And this is the code:

_manager = new RedmineManager(host, username, password);
var files = manager.GetObjects<Redmine.Net.Api.Types.File>(new NameValueCollection() { { RedmineKeys.PROJECT_ID, "5" } });_

"5" is the project id i'm using to test this functionality.

Thanks for help.
